src/rgw/rgw_op.cc

```diff
@@ -6356,3 +6356,117 @@ int RGWHandler::error_handler(int err_no, string *error_content) {
   // This is the do-nothing error handler
   return err_no;
 }
+
+
+void RGWPutBucketPolicy::send_response()
+{
+  if (op_ret) {
+    set_req_state_err(s, op_ret);
+  }
+  dump_errno(s);
+  end_header(s);
+}
+
+int RGWPutBucketPolicy::verify_permission()
+{
+  if (!verify_bucket_permission(s, rgw::IAM::s3PutBucketPolicy))
+
+    return -EACCES;
+
+  return 0;
+}
+
+int RGWPutBucketPolicy::get_params()
+{
+  if (s->length) {
+    datalen = atoll(s->length);
+    data.reset(new char[datalen + 1]);
```

Comment:
in general amazon restricts all xml input to a few thousand entities since a acl for eg. allows only 100 grants or so, RGW's older acls did not (so one could potentially write a 50G xml size in a putacl and make rgw return a ENOMEM)